torch/csrc/api/include/torch/detail/TensorDataContainer.h

```diff
@@ -0,0 +1,272 @@
+#pragma once
```

Comment:
Sometimes it is nicer to split declarations and definitions into .h and .cpp files. i.e., definitions go into .h files and declarations go into .cpp files. There are probably a lot of considerations about this that I am missing, but here are some pros and cons:

- (pro) compilation speeds: when one is making changes to TensorDataContainer.cpp, they don't need to recompile every file that that TensorDataContainer.h is included in
- (pro) modularity: Staring at the .h file gives a good sense of the interface 
- (con) Need to add a new file to the build system. Probably not the worse thing in the world

Reply:
Yes I think it's a good idea to split it into .h and .cpp files, although when I tried to do this I ran into a lot of "undefined reference" errors that I am still trying to find the reason for. I will attempt the split again when I implement the multidim support for the `at::tensor` constructor.

Reply:
Thanks a lot for the tip! I am gonna try that again in the next patch :/